src/net/socket.rs

```diff
@@ -156,10 +156,18 @@ impl Socket {
             }
         }
 
-        // Finally check for idle clients
+        // Check for idle clients
         if let Err(e) = self.handle_idle_clients(time) {
             error!("Encountered an error when sending TimeoutEvent: {:?}", e);
         }
+
+        // Finally send heartbeat packets to connections that require them
+        if let Err(e) = self.send_heartbeat_packets(time) {
```

Comment:
As discussed, I prefer to have this optional.

Reply:
Sure, should it be enabled or disabled by default?

Reply:
Made heartbeats disabled by default, and the config value is now an `Option<Duration>`, where `None` disables and `Some(Duration)` enables heartbeats.